### PR TITLE
show assignee image

### PR DESCRIPTION
**Pull request by MayurLund**
Issue https://github.com/Real-Dev-Squad/website-dashboard/issues/429

Overview : Made changes to function of createSuggestionList wherein destructured the picture property and extracted the url and put it into an src attribute of img tag.

Specifically, the function now handles user profile data that contains a picture property. The goal of this modification is to create a list of user suggestions, and for each suggestion, the user's profile picture is displayed as an image using an  tag alongside username. If profile url is not available then no profile pic as default to be shown.

Tests are yet to be written as I just had to display the image alongside username I just took one more property from user object i.e ->profile url. I have tested it locally by the process of reverse proxying.

Before :
screen-capture (3)

After :
screen-capture (4)

Included the Test for it too : 
image

### CODE REVIEW

**Review by kotesh-arya — DISMISSED** [review]
Looks good to Merge 👍🏻

**Review by Pratiyushkumar — COMMENTED** [review]
can we also include tests and test coverage please

**Comment by MayurLund** [comment]
> can we also include tests and test coverage please
@Pratiyushkumar As I just had to display the image alongside username I just took one more property from user object i.e ->profile url. I have tested it locally by the process of reverse proxying. I will create a ticket for test.

**Review by sahsisunny — DISMISSED** [review]

**Review by kotesh-arya — APPROVED** [review]

**Review by sahsisunny — APPROVED** [review]

**Comment by MayurLund** [comment]
> > can we also include tests and test coverage please
> 
> @Pratiyushkumar As I just had to display the image alongside username I just took one more property from user object i.e -profile url. I have tested it locally by the process of reverse proxying. I will create a ticket for test.

Hi @Pratiyushkumar, I have included tests now.

File: __tests__/tasks/profile-picture.test.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    //Trigger the input event manually
+    for (let i = 0; i < name.length; i++) {
+      await inputElement.type(name[i], { delay: 100 });
+    }
+
+    await page.waitForSelector('#suggested-users-container');
+
+    await page.waitForTimeout(2000);
+    const imgSrc = await page.$eval('#list-items img', (img) => img.src);
+    console.log(imgSrc);
```
Comment (Pratiyushkumar) [inline __tests__/tasks/profile-picture.test.js:66]: why are we doing console.log()??? any specific reason??

File: __tests__/tasks/profile-picture.test.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    for (let i = 0; i < name.length; i++) {
+      await inputElement.type(name[i], { delay: 100 });
+    }
+
+    await page.waitForSelector('#suggested-users-container');
+    await page.waitForTimeout(2000);
+
+    const imgSrc = await page.$eval('#list-items img', (img) => img.src);
+    console.log(imgSrc);
+    const expectedImageFilename = 'No-profile-pic.jpg';
```
Comment (Pratiyushkumar) [inline __tests__/tasks/profile-picture.test.js:89]: here also??

File: __tests__/tasks/profile-picture.test.js (hunk 1 above)
Comment (MayurLund) [reply __tests__/tasks/profile-picture.test.js]: I was verifying previously whether the urls are coming from the user object or not. I have removed it now.

File: __tests__/tasks/profile-picture.test.js (hunk 2 above)
Comment (MayurLund) [reply __tests__/tasks/profile-picture.test.js]: Thanks for pointing it out. I have removed it now.

**Review by Pratiyushkumar — APPROVED** [review]
LGTM

File: __tests__/tasks/profile-picture.test.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    const buttonId = 'statusId';
+    const button = await page.$(`#${buttonId}`);
+    await button.click();
+    await page.waitForSelector('select.input:not(.notEditing)');
+    await page.select('select#status', 'ASSIGNED');
+  });
+
+  afterAll(async () => {
+    await browser.close();
+  });
```
Comment (prakashchoudhary07) [inline __tests__/tasks/profile-picture.test.js:51]: Nit: we should keep this above first test

File: __tests__/tasks/profile-picture.test.js [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    await page.waitForSelector('#suggested-users-container');
+
+    await page.waitForTimeout(2000);
+    const imgSrc = await page.$eval('#list-items img', (img) => img.src);
+    const expectedImageFilename =
+      'https://res.cloudinary.com/imgprc/image/upload/v1688145505/profile/tu4zVmwFX91e9oEuSWLg/woqilcqdbp0cudcexkem.jpg';
+    expect(imgSrc.endsWith(expectedImageFilename)).toBe(true);
+  });
```
Comment (prakashchoudhary07) [inline __tests__/tasks/profile-picture.test.js:69]: How are we getting the actual URL here?
Are we making actual API call and checking?
Tests should not be doing actual network calls.

**Review by prakashchoudhary07 — CHANGES_REQUESTED** [review]

File: __tests__/tasks/profile-picture.test.js (hunk 4 above)
Comment (MayurLund) [reply __tests__/tasks/profile-picture.test.js:69]: No, I have created mock data and in that mocked data there is url and when the api is getting hit I  am mocking the api response and sending that mock data and checking the value of url present in mock data equal to src element of img tag.
image

File: __tests__/tasks/profile-picture.test.js (hunk 3 above)
Comment (MayurLund) [reply __tests__/tasks/profile-picture.test.js:51]: Sure, will keep in mind.

**Review by prakashchoudhary07 — APPROVED** [review]

**Pull request merged**